src/utils/product-filter.jsx

```diff
@@ -0,0 +1,70 @@
+export const compositeFilter =
+  (state, ...filterFunctions) =>
+  (products) => {
+    // console.log(products);
+    return filterFunctions.reduce(
+      (acc, currFunction) => currFunction(state, acc),
+      products
+    );
+  };
+
+export const filterRatings = (state, products) => {
+  // console.log("inside", products);
+  const temp =
+    state.ratings === 0
+      ? products
+      : products.filter(({ ratings }) => ratings >= state.ratings);
+  // console.log("temp", temp);
+  return temp;
+};
+
+export const filterPrice = (state, products) => {
+  // console.log("inside", products);
+  const temp =
+    state.price.min === 0 && state.price.max === 5000
+      ? products
+      : products.filter(
+          ({ discountedPrice }) =>
+            discountedPrice >= state.price.min &&
+            discountedPrice <= state.price.max
+        );
+  // console.log("temp", temp);
```

Comment:
remove all the commented console.log's, it's not a good to have commented codes's and console log's in production code as well.